src/main/java/chess/domain/piece/Piece.java

```diff
@@ -0,0 +1,55 @@
+package chess.domain.piece;
+
+import chess.domain.piece.position.Path;
+import chess.domain.piece.position.PiecePosition;
+
+import java.util.List;
+
+public class Piece {
+
+    protected final Color color;
+    protected final PiecePosition piecePosition;
+    protected final PieceMovement pieceMovement;
+
+    public Piece(final Color color, final PiecePosition piecePosition, final PieceMovement pieceMovement) {
+        this.color = color;
+        this.piecePosition = piecePosition;
+        this.pieceMovement = pieceMovement;
+    }
+
+    /**
+     * @throws IllegalArgumentException 이동할 수 없는 경로가 들어온 경우
+     */
+    public List<PiecePosition> waypoints(final PiecePosition destination, final Piece nullablePiece) throws IllegalArgumentException {
+        return pieceMovement.waypoints(color, path(destination), nullablePiece);
+    }
```

Comment:
```suggestion
    public List<PiecePosition> waypoints(final PiecePosition destination, final Piece nullablePiece) {
        return pieceMovement.waypoints(color, path(destination), nullablePiece);
    }
```

runtimeException인데 따로 명시해준 이유가 있나요?

Reply:
구현하는 사람 입장에서 잘못되었을때 이러한 예외를 반드시 던져야 한다는 것을 명시하기 위함이었습니다.

Reply:
runtimeException의 장점을 모두 무시하는 행위 및 가독성을 심하게 해치는 방법이라고 생각합니다
예외를 반드시 던져야한다면 checked exception을 던지는게 바람직할것 같고요